crates/external_services/src/http_client.rs

```diff
@@ -0,0 +1,182 @@
+use common_utils::{consts, errors::CustomResult, request::Request};
+use hyperswitch_interfaces::{errors::HttpClientError, types::Proxy};
+use request::{HeaderExt, RequestBuilderExt};
+use router_env::{instrument, logger, tracing};
+/// client module
+pub mod client;
+/// metrics module
+pub mod metrics;
+/// request module
+pub mod request;
+use std::{error::Error, time::Duration};
+
+use common_utils::request::RequestContent;
+pub use common_utils::request::{ContentType, Method, RequestBuilder};
+use error_stack::ResultExt;
+
+#[allow(missing_docs, missing_debug_implementations)]
```

Comment:
Nit: We don't have to allow `missing_debug_implementations` for a function, clippy shouldn't complain about it.